TAOK3.py

```diff
@@ -0,0 +1,376 @@
+
+    The output file also has fastq extension.
+
+    The output file is saved in the 'fastq_filtrator_results' directory.
+
+    If 'fastq_filtrator_results' directory doesn't exist the program creates it
+    in a current directory.
+
+    Without full names use arguments in a certain order
+    Example: filter_fastq(input_path, output_filename, (0,100), (0,200), 0)
+           # filter_fastq(input_path, output_filename, gc_bounds=(0,100),
+             length_bounds=(0,200), quality_threshold=0)
+
+    In case of changing only one argument, provide its full name!
+    Example: filter_fastq(input_path, output_filename,
+                          length_bounds=(50, 100))
+
+    Arguments:
+
+    - input_path(str): the path to the file
+
+    - output_filename(str): the name for output file with obtained result
+    By default output_filename=None
+    Without output_filename argument the output file is named as input file
+    Name without fastq extention is acceptible.
+    Example: output_filename='result'  # 'result.fastq'
+             output_filename='result.fastq'
+
+    - gc_bounds(tuple or int or float): contain min and max GC-content bounds
+    By default gc_bounds=(0,100)
+    If input contains one number the function accepts it as a maximum bound
+    Examples: gc_bounds=(20,40)
+              gc_bounds=40  # (0,40)
+
+    - length_bounds(tuple or int or float): contain mini and max length bounds
+    By default length_bounds=(0,4294967296)
+    If input contains one number the function accepts it as a maximum bound
+    Examples: length_bounds=(10,90)
+              length_bounds=90  # (0,90)
+
+    - quality_threshold(int or float): the quality threshold of the interest
+    By default quality_threshold=0
+    Examples: quality_threshold=10
+
+    There are three functions that are used in the main function:
+
+    - is_in_gc_bounds(seq_record, gc_bounds):
+    Check if the sequence falls in the range of GC-content bounds
+
+    - is_in_length_bounds(seq_record, length_bounds):
+    Check if the sequence falls in the range of length bounds
+
+    - is_above_quality_threshold(seq_record, quality_threshold):
+    Check if the mean of quality values exceeds the quality threshold
+
+    Return:
+    - file: file with fastq extension containing selected fragments.
+
+    For more information please see README
+
+    """
+    if type(length_bounds) is int:
+        length_bounds = 0, length_bounds
+    if type(gc_bounds) is int or type(gc_bounds) is float:
+        gc_bounds = 0, gc_bounds
+    sequences = SeqIO.parse(input_path, "fastq")
+    good_reads = []
+    for seq_record in sequences:
+        if (is_in_gc_bounds(seq_record, gc_bounds) and
+            is_in_length_bounds(seq_record, length_bounds) and
+            is_above_quality_threshold(seq_record, quality_threshold)):
+            good_reads += [seq_record]
+    if len(good_reads) == 0:
+        raise ValueError('There are no sequences suited to requirements')
```

Comment:
Кажется, что лучше просто вывести предупреждение в stdout о конкретном файле. Было бы неплохо прогонять фильтр по нескольким файлам в цикле... Но если хоть в одном из них не будет хорошей последовательности, то цикл остановится с ошибкой - неприятно

```suggestion
    if len(good_reads) == 0:
        print(f'In {input_path} there is no sequences suited to requirements')
```